src/Rule/Each.php

```diff
@@ -20,15 +21,19 @@
  * Validates an array by checking each of its elements against a set of rules.
  */
 #[Attribute(Attribute::TARGET_PROPERTY | Attribute::IS_REPEATABLE)]
-final class Each implements SerializableRuleInterface, BeforeValidationInterface, SkipOnEmptyInterface
+final class Each implements
+    SerializableRuleInterface,
+    BeforeValidationInterface,
+    SkipOnEmptyInterface,
+    PropagateOptionsInterface
 {
     use BeforeValidationTrait;
     use RuleNameTrait;
     use SkipOnEmptyTrait;
 
     public function __construct(
         /**
-         * @var iterable<RuleInterface>
+         * @var iterable<BeforeValidationInterface|RuleInterface|SkipOnEmptyInterface>
```

Comment:
Does the rule always implement all these interfaces?

Reply:
Yes. It was added specifically for Psalm. Maybe there is a simpler way?